### DISCUSSION TITLE
How about deref-defs?

### DISCUSSION THREAD

**Discussion by eval** [borkdude/deflet#3] [Ideas]
'eval-stepping' through a bunch of `@(def x y)` would directly show the value of the bounded var.
Is that something that is viable/wanted to support in this library as well?

**Comment by borkdude** [comment]
Do you mean that in the REPL you would eval `@(def x y)` but it would still compile to `(let [x y])`?

**Reply by eval** [reply]
Indeed.

**Reply by borkdude** [reply]
Maybe we could have a better shortcut for this. E.g.:

```
(require '[borkdude.deflet :refer [deflet letvar]]

(deflet
  (letvar x 10) ;;=> 10, but defines var x
)
```

where `letvar` inside `deflet` expands into a normal let binding.

I also considered just using `let` or `var` but this clashes with the core var/special form.

**Reply by borkdude** [reply]
A shorter name would be nice though

**Reply by borkdude** [reply]
Maybe:

``` clojure
(require '[borkdude.deflet :refer [deflet bind]]

(deflet
  (bind x 10) ;;=> 10, but defines var x
)
```

or so. Other name ideas welcome...

Maybe: `(with x 10)`

**Reply by borkdude** [reply]
Yeah, I think I like `bind`

**Reply by eval** [reply]
Apart from this issue - why step away from `def`?

Regarding this issue: I like the variants on `def` (defp), e.g. 'def->' ("def through"). And there will be a `deft` ("def tap") in the future... ;)

**Reply by borkdude** [reply]
Maybe `defd`, short for `def-deref`?

**Reply by eval** [reply]
Sounds good!

**Reply by borkdude** [reply]
I think why I don't like `@(def x 10)` is that you will have lots of derefs in your code, which looks kind of ugly

**Reply by borkdude** [reply]
Although for nbb we will get:

```
(defpd y (p/delay 100 (+ x 1)))
```

which is kind of ugly...

**Reply by borkdude** [reply]
I have a branch here with `bind`: https://github.com/borkdude/deflet/tree/bind
Let's sleep on it for a night and feel free to take it for a spin

**Reply by eval** [reply]
Thanks! Will try it.

**Reply by borkdude** [reply]
You could of course also insert a comment form:

``` clojure
(deflet
  (def x 10)
  (comment
    x
    (tap> x))
  (inc x))
```

**Reply by borkdude** [reply]
or:

``` clojure
  (deflet
    (def x 10)
    #_(do
        x
        (tap> x))
    (inc x))
```

**Reply by borkdude** [reply]
And also you could just evaluate the symbol `x` in `(def x 10)` after you've def-ed `x` in your REPL.

**Reply by eval** [reply]
Yeah, this doesn't need any additions indeed 👍🏻
